exercise/views.py

```diff
@@ -72,6 +73,14 @@ def get(self, request, *args, **kwargs):
         students = [self.profile]
         all_enroll_data = None
 
+        userprofile = self.request.user.userprofile
```

Comment:
Could the template be dynamically selected in a method like `get_template_names` instead of in the `get` method? Switching the template feels fishy anyway.

https://docs.djangoproject.com/en/2.2/ref/class-based-views/mixins-simple/#django.views.generic.base.TemplateResponseMixin.get_template_names